src/lib/index.ts

```diff
@@ -67,100 +71,96 @@ export class GoogleToken {
    *
    * @param callback The callback function.
    */
-  public getToken(callback: (err: Error|null, token?: string|null) => void):
-      void {
-    const handleJSONKey = (err: Error, key: string) => {
-      if (err) {
-        callback(err);
-        return;
-      }
-      try {
-        const body = JSON.parse(key);
-        this.key = body.private_key;
-        this.iss = body.client_email;
-      } catch (e) {
-        callback(e);
-        return;
-      }
+  getToken(): Promise<string>;
+  getToken(callback: (err: Error|null, token?: string|null) => void): void;
+  getToken(callback?: (err: Error|null, token?: string|null) => void):
```

Comment:
Not for this change, but can `token` be `null` when it's already optional?

Reply:
It really can be :(  It's derived from this.token, which is nulled out in several places.  This is actually pretty consistent across all of these libraries we adopted.

Reply:
That means users should check both the error and whether the token returned is null? That's very unfortunate.

If so, should the return type of the promise-version of `getToken()` be `Promise<string|null>` above?

Reply:
After taking a second look...  I was able to get this right.  I just throw down in `requestToken` if the call comes back, and it happens to be null (which would be unexpected anyway).  Great feedback :)

Reply:
And after a third look lol...  it can be null!  There are actually tests that were covering this situation.  I updated the promise signature to be `Promise<string|null>`.  Thanks!